src/sonic-eventd/rsyslog_plugin/syslog_parser.cpp

```diff
@@ -19,9 +19,35 @@ bool SyslogParser::parseMessage(string message, string& eventTag, event_params_t
         }
         // found matching regex
         eventTag = m_regexList[i]["tag"];
-        transform(params.begin(), params.end(), matchResults.begin() + 1, inserter(paramMap, paramMap.end()), [](string a, string b) {
-            return make_pair(a,b);
-        });
+	// check params for lua code
+	for(long unsigned int j = 0; j < params.size(); j++) {
+            auto delimPos = params[j].find(':');
+	    string resultValue = matchResults[j + 1].str();
+            if(delimPos != string::npos) { // have to execute lua script
```

Comment:
Can you put the most common scenario of == NULL first.
Apparently the compiler expects the first case to be common and optimize the binary compiled code by avoiding a jump for most common case.

Reply:
Done